pymc3/sampling.py

```diff
@@ -903,7 +903,18 @@ def _iter_sample(
     try:
         step.tune = bool(tune)
         if hasattr(step, 'reset_tuning'):
-            step.reset_tuning()
+            if isinstance(step, CompoundStep):
```

Comment:
The `CompoundStep` already has a method for custom `reset_tuning` logic: https://github.com/pymc-devs/pymc3/blob/master/pymc3/step_methods/compound.py#L70

Reply:
Moved the logic inside Metropolis' reset_tuning()